### DISCUSSION TITLE
Slowness issue when using Supavisor connection pooling with Prisma

### DISCUSSION THREAD

**Discussion by MatteoGauthier** [supabase/supabase#28141] [Questions]
Hey Supabase community, I've set up Supavisor on my project but it turns out that the request is abnormally slow with Prisma when I use supavisor. I compared the time it took to make a request with `console.time` and the result was clear.

The tests are done locally with a remote postgres database hosted on Supabase with Supavisor enabled.

Here is my connection string in my `.env`

```bash
# Slow setup for Supabase with Supavisor
DATABASE_URL="postgres://postgres.project_id:password@aws-0-eu-central-1.pooler.supabase.com:6543/postgres?pgbouncer=true&connection_limit=1"
DIRECT_DATABASE_URL="postgres://postgres.project_id:password@aws-0-eu-central-1.pooler.supabase.com:5432/postgres"

# Fast setup for Supabase without using Supavisor
DATABASE_URL="postgres://postgres.project_id:password@aws-0-eu-central-1.pooler.supabase.com:5432/postgres"
DIRECT_DATABASE_URL="postgres://postgres.project_id:password@aws-0-eu-central-1.pooler.supabase.com:5432/postgres"

```

```prisma
// schema.prisma

datasource db {
    provider  = "postgresql"
    url       = env("DATABASE_URL")
    directUrl = env("DIRECT_DATABASE_URL")
}
```

```ts
    console.time("totalStudents");
    const totalStudents = await db.user.count({
      where: {
        role: "STUDENT",
      },
    });
    console.timeEnd("totalStudents");

```

With Supavisor : `291.246ms`
Without Supavisor (same connection string in `DATABASE_URL`and `DIRECT_DATABASE_URL`) : `52.009ms`

Does someone can help me ?

Have a nice day  :wink:

**Comment by m3hari** [comment]
Same. My only explanation is Supervisor is a shared resource and performance is inconsistent

**Comment by devstein** [comment]
We are seeing the same performance difference (w/ vs w/o Supavisor) at @dosu-ai and haven't been able to root cause

**Comment by therealpurplemana** [comment]
Same
